stor/dx.py

```diff
@@ -387,34 +1030,43 @@ def exists(self):
             return True
         except (NotFoundError, ValueError):
             pass
-        # otherwise we could be a directory, so try to grab first
-        # file/subfolder
-        if self._is_folder():
-            try:
-                self.list(limit=1)
-                return True
-            except NotFoundError:
-                return False
-        return False
+        # otherwise we could be a directory, so try to listdir folder
+        # note list doesn't error on non-existent folder and cannot be used here
+        try:
+            self.listdir()
+            return True
+        except NotFoundError:
+            return False
+
+    def getsize(self):
+        if not self.resource:
+            return self.stat()['dataUsage']*1e9
+        else:
+            return self.stat()['size']
 
+    @_propagate_dx_exceptions()
```

Comment:
cool that this works - I'm a little skeptical of using contextlib2 just to make this work rather than just using it in a with statement - but I guess clearer that way

Reply:
Yeah, the thing is when everything under the function has to be inside the with statement, might as well go with the annotator.